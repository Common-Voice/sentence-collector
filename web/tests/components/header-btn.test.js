import { shallow } from 'enzyme';
import React from 'react';

import HeaderBtn from '../../src/components/header-btn';

const clickHandler = jest.fn();

test('Header button should be rendered correctly when closed', () => {
  let component = shallow(<HeaderBtn isOpen={false} handleClick={clickHandler} />);
  expect(component).toMatchSnapshot();
});

test('Header button should be rendered correctly when open', () => {
  let component = shallow(<HeaderBtn isOpen={true} handleClick={clickHandler} />);
  expect(component).toMatchSnapshot();
});

test('Header button should close header on click', () => {
  let component = shallow(<HeaderBtn isOpen={true} handleClick={clickHandler} />);
  component.find('button').simulate('click');
  expect(clickHandler.mock.calls.length).toEqual(1);
});
