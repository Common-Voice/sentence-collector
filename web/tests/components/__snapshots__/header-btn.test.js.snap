// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header button should be rendered correctly when closed 1`] = `
<button
  className="header-btn header-btn-closed"
  onClick={[Function]}
>
  <div
    className="header-btn-arrow header-btn-arrow-right"
  />
</button>
`;

exports[`Header button should be rendered correctly when open 1`] = `
<button
  className="header-btn header-btn-open"
  onClick={[Function]}
>
  <div
    className="header-btn-arrow"
  />
</button>
`;
